packages/network/README.md

```diff
@@ -29,44 +29,36 @@ where the user code is really accepting an inbound IBC *Channel*.
 
-Vats which live inside a solo machine are able to use traditional networking layers, like TCP, HTTP, and WebSockets. This enables them to communicate with e.g. browser-side UI frontends that use WebSockets to query the vat for order status. These connections do not have to follow normal ocap rules, so vat code which accepts them must provide its own authentication and access control protections.
-
-Solo machines may be able to talk to chains and vice versa using specialized protocols. This will be used by CapTP to provide ocap messaging between these heterogeneous machine types.
-
-## The agoric-sdk User Local Port
-
-Each user of the Agoric testnet gets a few personal IBC listening ports. You can access these `Port` objects in the `home.ibcport` array, and you can learn their local address by calling something like `E(home.ibcport[0]).getLocalAddress()`, which will give you a local address like `/ibc-port/portbvmnfb`.
-
-This is currently the only way for user code to get an IBC `Port`, though non-IBC ports can be allocated using the local `home.network` object.  This is an advanced use case, to be documented later.
-
 ## Connecting to a Remote Port
 
 To establish a connection, you must start with a local `Port` object, and you must know the name of the remote endpoint. The remote endpoint will have a name like `/ibc-hop/$HOPNAME/ibc-port/$PORTNAME/ordered/$VERSION` (where `ibc-hop`, `ibc-port` and `ordered` are literal strings, spelled just like that, but `$HOPNAME`, `$PORTNAME`, and `$VERSION` are placeholders for arbitrary values that will vary from one endpoint to another).
 
-You must also prepare a `ConnectionHandler` object to manage the connection you're about to create. This has a number of methods which will be called when the things happen to the connection, including packets arriving. This is described in [Receiving Data](#receiving-data).
+Ports can be obtained from the `PortAllocator` in the `network` vat. Call `allocateCustomIBCPort()`, `allocateICAControllerPort()`, or `allocateICQControllerPort()` to get a port who's naming is scoped to a specific application protocol. _Binding transfer ports is not supported through network, but users are able to use the `transfer` vat to register an application to listen to or intercept ics-20 transfer packets._ As more network and application protocols emerge, we expect to support additional `PortAllocator` methods and naming conventions.
```

Comment:
```suggestion
Ports can be obtained from the `PortAllocator` in the `network` vat. Call `allocateCustomIBCPort()`, `allocateICAControllerPort()`, or `allocateICQControllerPort()` to get a port whose naming is scoped to a specific application protocol. _Binding transfer ports is not supported through network, but users are able to use the `transfer` vat to register an application to listen to or intercept ics-20 transfer packets._ As more network and application protocols emerge, we expect to support additional `PortAllocator` methods and naming conventions.
```

Reply:
It would be valuable to elaborate on the differences and implications of an automatically assigned vs a custom/vanity port. My understanding is that port allocator effectively prevents collisions, however this is relaxed for vanity/custom ports which are prefixed `custom-`.

Users should understand that automatic assignments protect them against squatting. However, they do not get the same guarantees with ports allocated with the `custom` prefix. Thoughts?

Reply:
If a caller binds a vanity port by calling `allocateCustomIBCPort('vain')`, then every subsequent attempt to `allocateCustomIBCPort('vain')` will return a rejected promise.  Rather than avoiding the collision automatically, it leaves the burden to the caller to respond appropriately to rejections.  That's exactly what would happen if `bindPort` is used directly.